### PR TITLE
feat: sdk7 tween component

### PR DESCRIPTION
**Pull request by pravusjif**
Implemented Tween component from [SDK7 Smooth Movement Shape Up Project](https://www.notion.so/decentraland/SDK7-Smooth-Movement-1f86fbfaeb084d57999456dca1facec7)

Related PRs:
* js-sdk-toolchain: https://github.com/decentraland/js-sdk-toolchain/pull/753
* protocol: https://github.com/decentraland/protocol/pull/175

### <samp>🤖 Generated by Copilot at 9edc801</samp>

Added support for tween animations in the ECS7 plugin. Updated the `@dcl/protocol` dependency and refactored the `ECSTransformHandler` class. Created new classes, files and tests for the `PBTween` and `PBTweenState` components and their handlers and serializers.

### CODE REVIEW

**Comment by pravusjif** [comment]
I'll update the protocol package after this PR is approved for merging (and after the related [protocol PR](https://github.com/decentraland/protocol/pull/175) is merged).

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Tween/ECSTweenHandler.cs
Comment (Kinerius) [inline unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Tween/ECSTweenHandler.cs:213]: To avoid this huge switch we could just make a `Dictionary<EasingFunction, Ease>` and use that as a map, then we could replace this switch with a TryGet, WDYT?

**Review by Kinerius — APPROVED** [review]
Nice!

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Tween/ECSTweenHandler.cs
Comment (pravusjif) [reply unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Tween/ECSTweenHandler.cs]: Done!

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Tween/ECSTweenHandler.cs
Comment (popuz) [inline unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Tween/ECSTweenHandler.cs:183]: I think @Kinerius idea was to have it directly in the variable declaration, which together with `using static` can be a quite readable map
 
```
using static DCL.ECSComponents.EasingFunction;
using static DG.Tweening.Ease;

private readonly Dictionary<EasingFunction, Ease> easingFunctionsMap = new()
    {
        [TfEaseinsine] = InSine,
        [TfEaseoutsine] = OutSine,
        [TfEasesine] = InOutSine,
        [TfEaseinquad] = InQuad,
        [TfEaseoutquad] = OutQuad,
        [TfEasequad] = InOutQuad,
        [TfEaseinexpo] = InExpo,
        [TfEaseoutexpo] = OutExpo,
        [TfEaseexpo] = InOutExpo,
        [TfEaseinelastic] = InElastic,
        [TfEaseoutelastic] = OutElastic,
        [TfEaseelastic] = InOutElastic,
        [TfEaseinbounce] = InBounce,
        [TfEaseoutbounce] = OutBounce,
        [TfEasebounce] = InOutBounce,
        [TfEaseincubic] = InCubic,
        [TfEaseoutcubic] = OutCubic,
        [TfEasecubic] = InOutCubic,
        [TfEaseinquart] = InQuart,
        [TfEaseoutquart] = OutQuart,
        [TfEasequart] = InOutQuart,
        [TfEaseinquint] = InQuint,
        [TfEaseoutquint] = OutQuint,
        [TfEasequint] = InOutQuint,
        [TfEaseincirc] = InCirc,
        [TfEaseoutcirc] = OutCirc,
        [TfEasecirc] = InOutCirc,
        [TfEaseinback] = InBack,
        [TfEaseoutback] = OutBack,
        [TfEaseback] = InOutBack,
    };
```

Comment (pravusjif) [reply unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Tween/ECSTweenHandler.cs]: Cool thanks! now I implemented it that way.

File: unity-renderer/Assets/DCLPlugins/ECS7/Systems/TweenSystem/ECSTweenSystem.cs
Comment (popuz) [inline unity-renderer/Assets/DCLPlugins/ECS7/Systems/TweenSystem/ECSTweenSystem.cs:37]: I would suggest to hierarchically split it into 2 functions for better readability
```
public void Update()
        {
            var tweenComponentGroup = tweenInternalComponent.GetForAll();
            
            for (var i = 0; i < tweenComponentGroup.Count; i++)
                UpdateTweenComponentModel(tweenComponentGroup[i]);
        }

        private void UpdateTweenComponentModel(KeyValueSetTriplet<IParcelScene, long, ECSComponentData<InternalTween>> tweenComponentGroup)
        {
            var tweenStatePooledComponent = tweenStateComponentPool.Get();
            var tweenStateComponentModel = tweenStatePooledComponent.WrappedComponent.Model;
            IParcelScene scene = tweenComponentGroup.key1;
            if (!componentsWriter.TryGetValue(scene.sceneData.sceneNumber, out var writer))
                return;

            long entityId = tweenComponentGroup.key2;
            InternalTween model = tweenComponentGroup.value.model;

            if (model.removed)
            {
                model.tweener.Kill();
                writer.Remove(entityId, ComponentID.TWEEN_STATE);
                return;
            }

            float currentTime = model.tweener.ElapsedPercentage();
            if (currentTime.Equals(1f) && model.currentTime.Equals(1f))
                return;

            if (model.playing)
            {
                tweenStateComponentModel.State =
                    currentTime.Equals(1f) ? TweenStateStatus.TsCompleted : TweenStateStatus.TsActive;
                UpdatePlayingTweenComponentModel(tweenStateComponentModel, currentTime, scene, entityId, model);
            }
            else
            {
                tweenStateComponentModel.State = TweenStateStatus.TsPaused;
            }

            writer.Put(entityId, ComponentID.TWEEN_STATE, tweenStatePooledComponent);

            UpdateTransformComponent(scene, entityId, model.transform, writer);

            model.currentTime = currentTime;
            tweenInternalComponent.PutFor(scene, entityId, model);
        }

        private void UpdatePlayingTweenComponentModel(PBTweenState tweenStateComponentModel, float currentTime,
            IParcelScene scene, long entityId, InternalTween model)
        {
            tweenStateComponentModel.CurrentTime = currentTime;
            scene.entities.TryGetValue(entityId, out IDCLEntity entity);
            switch (model.tweenMode)
            {
                case PBTween.ModeOneofCase.Move:
                    sbcInternalComponent.SetPosition(scene, entity, model.transform.position);
                    break;
                case PBTween.ModeOneofCase.Rotate:
                case PBTween.ModeOneofCase.Scale:
                    sbcInternalComponent.OnTransformScaleRotationChanged(scene, entity);
                    break;
            }
        }
```

**Review by popuz — APPROVED** [review]
Nice bridge between SDK and DOTween 💪 I left mostly cosmetical comments

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Tween/ECSTweenHandler.cs
Comment (pravusjif) [reply unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Tween/ECSTweenHandler.cs:161]: Uhmm yeah but since these component handlers only have 1 instance, I'm just using all readonly private stuff

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Tween/ECSTweenHandler.cs
Comment (pravusjif) [reply unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Tween/ECSTweenHandler.cs]: Oh yeah, missed that

Comment (pravusjif) [reply unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Tween/ECSTweenHandler.cs]: Done!

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Tween/ECSTweenHandler.cs
Comment (pravusjif) [reply unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Tween/ECSTweenHandler.cs]: Done!

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Tween/ECSTweenHandler.cs
Comment (pravusjif) [reply unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Tween/ECSTweenHandler.cs]: Done!

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Tween/ECSTweenHandler.cs
Comment (pravusjif) [reply unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Tween/ECSTweenHandler.cs]: Done!

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Utils/ProtoConvertUtils.cs
Comment (pravusjif) [reply unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Utils/ProtoConvertUtils.cs]: Done!

File: unity-renderer/Assets/DCLPlugins/ECS7/Systems/TweenSystem/ECSTweenSystem.cs
Comment (pravusjif) [reply unity-renderer/Assets/DCLPlugins/ECS7/Systems/TweenSystem/ECSTweenSystem.cs:38]: Done!

**Pull request merged**
